syntax = "proto2";

// Client Messages


message RequestNewEpisode {


  optional bytes init_file = 1;

}

message EpisodeStart {


  optional int32 start_index =1;
  optional int32 end_index =2;


}


message Control {

  optional float steer = 1;
  optional float gas = 2;


}


// Server messages



message Scene {

  //optional int32 number_of_cameras = 1;
  optional bytes positions = 1;
  //optional bytes projection_matrices = 3;

}

message EpisodeReady{

    optional bool ready =1;
}






message Reward {

  optional float player_x = 1;
  optional float player_y = 2;
  optional float speed = 3;
  optional float collision_gen = 4;
  optional float collision_ped = 5;
  optional float collision_car = 6;
  optional float road_intersect = 7;
  optional float sidewalk_intersect = 8;
  optional float acceleration_x = 9;
  optional float acceleration_y = 10;
  optional float acceleration_z = 11;
  optional int32 platform_timestamp = 12;
  optional int32 game_timestamp = 13;
  optional float ori_x = 14;
  optional float ori_y = 15;
  optional float ori_z = 16;
  optional bytes image_sizes = 17;
  optional bytes finalImage_sizes = 18;
  optional bytes depth_sizes = 19;
  optional bytes semanticSeg_sizes = 20;
  optional bytes images = 21;
  optional bytes finalImages = 22;
  optional bytes depths = 23;
  optional bytes semanticSegs = 24;

}
